### DISCUSSION TITLE
Runtime wiring validation

### DISCUSSION THREAD

**Discussion by pmhsfelix** [graphql-java/graphql-java#2692] [Q&A] [answered]
Is there a recommended way to validate a `RuntimeWiring` before it starts to be used? The use case is to enforce rules on the wiring model, such as checking if the top-level fetchers are of a given type.

IINM, a possible way to achieve this is to create a `SchemaDirectiveWiring` and then register it using the `directiveWiring` method. The `SchemaDirectiveWiring` implementation will then be called on all directive targets (e.g. fields), giving it a way to check what is already wired-up.
However, I wonder if there is a more recommended way of achieving this.

**Comment by bbakerman** [comment, marked as answer]
> The use case is to enforce rules on the wiring model, such as checking if the top-level fetchers are of a given type.
I am guessing you mean the DataFetchers (java code) are of a certain Java type?  (as oppossed to graphql types)

If that right then `SchemaDirectiveWiring` is the way to go.  If we had our time again, `SchemaDirectiveWiring` should be called `SchemaWiring` since it's really a callback that allows you to see each field more easily.

There is a another callback called `SchemaGeneratorPostProcessing` in this but it only gives you the schema and you would need to traverse it and the CodeRegistry etc.. which SchemaDirectiveWiring does for you.

**Reply by pmhsfelix** [reply]
Thanks. Yes, I did meant `DataFetchers` (i.e. java code) on my original question. I was afraid that `SchemaDirectiveWiring` was too connected to directive processing however your answer makes it clear that it isn't.
